openedx/core/djangoapps/content/search/migrations/0001_initial.py

```diff
@@ -0,0 +1,49 @@
+# Generated by Django 4.2.10 on 2024-04-02 04:55
+
+from django.db import migrations, models
+from opaque_keys.edx.django.models import LearningContextKeyField
+from opaque_keys.edx.locator import LibraryLocatorV2
+
+
+def populate_search_access(apps, schema_editor):
+    """
+    Adds SearchAccess rows for all existing CourseOverview and ContentLibrary instances.
+    """
+    if schema_editor.connection.alias != "default":
+        return
+
+    SearchAccess = apps.get_model("search", "SearchAccess")
+
+    CourseOverview = apps.get_model("course_overviews", "CourseOverview")
+    for course in CourseOverview.objects.only('id'):
+        SearchAccess.objects.create(context_key=course.id)
+
+    ContentLibrary = apps.get_model("content_libraries", "ContentLibrary")
+    for lib in ContentLibrary.objects.select_related("org").only("org", "slug"):
+        # ContentLibrary.library_key property is not available here, so we have to assemble it.
+        lib_key = LibraryLocatorV2(org=lib.org.short_name, slug=lib.slug)
+        SearchAccess.objects.create(context_key=lib_key)
```

Comment:
If we're concerned about the performance of this data migration, I can remove it and add a management command instead.

Reply:
Decided to remove the data migration ([ef89591](https://github.com/openedx/edx-platform/pull/34471/commits/ef89591755a07d69aa8a4934240cfaeb180ff745)): running [`reindex_studio` handles it for us](https://github.com/open-craft/edx-platform/blob/jill/content-search-perms/openedx/core/djangoapps/content/search/documents.py#L83-L88).